### PR TITLE
ZMQ lib change.

### PR DESCRIPTION
**Pull request by divyagayathri-hcl**
### Build Results:
divya@ddb4b377439b:/sonic/src/sonic-p4rt/sonic-swss-common$ bazel build $BAZEL_BUILD_OPTS ...
Starting local Bazel server and connecting to it...
INFO: Analyzed 4 targets (30 packages loaded, 544 targets configured).
INFO: Found 4 targets...
INFO: Elapsed time: 13.344s, Critical Path: 5.62s
INFO: 16 processes: 6 internal, 10 linux-sandbox.
INFO: Build completed successfully, 16 total actions

### Test Results:
divya@ddb4b377439b:/sonic/src/sonic-p4rt/sonic-swss-common$ bazel test $BAZEL_BUILD_OPTS --cache_test_results=no ...
INFO: Build option --cache_test_results has changed, discarding analysis cache.
INFO: Analyzed 4 targets (0 packages loaded, 544 targets configured).
INFO: Found 2 targets and 2 test targets...
INFO: Elapsed time: 1.086s, Critical Path: 0.60s
INFO: 3 processes: 1 internal, 2 linux-sandbox.
INFO: Build completed successfully, 3 total actions
//tests:saiaclschema_ut                                                  PASSED in 0.5s
//tests:status_code_util_test                                            PASSED in 0.5s

Executed 2 out of 2 tests: 2 tests pass.
INFO: Build completed successfully, 3 total actions
divya@ddb4b377439b:/sonic/src/sonic-p4rt/sonic-swss-common$

### CODE REVIEW

File: common/zmqclient.cpp [hunk 1]
```diff
@@ -77,17 +78,26 @@ void ZmqClient::connect()
         zmq_ctx_destroy(m_context);
     }
     
-    // ZMQ Client/Server are n:1 mapping, so need use PUSH/PULL pattern http://api.zeromq.org/master:zmq-socket
     m_context = zmq_ctx_new();
-    m_socket = zmq_socket(m_context, ZMQ_PUSH);
-    
+    if (m_oneToOneSync)
```
Comment (mint570) [inline common/zmqclient.cpp:82]: Don't change this.
We cannot upstream ZMQ_REQ type. It has limitation.

File: common/zmqclient.h [hunk 2]
```diff
@@ -12,7 +12,10 @@ namespace swss {
 class ZmqClient
 {
 public:
-    ZmqClient(const std::string& endpoint);
+    // If waitTimeMs is set to non-zero, it will enable one-to-one sync with the
```
Comment (mint570) [inline common/zmqclient.h:15]: We cannot upstream one-to-one sync mode.

Just update the comment to explain the waitTimeMs argument is for response path.

File: common/zmqclient.h [hunk 3]
```diff
@@ -35,6 +45,12 @@ class ZmqClient
 
     bool m_connected;
 
+    // Timeout in ms to wait for response message from the server.
+    // If this is set to zero, one-to-one sync is disabled.
+    uint32_t m_waitTimeMs;
+
+    bool m_oneToOneSync;
```
Comment (mint570) [inline common/zmqclient.h:52]: Remove this. We cannot upstream one-to-one sync

File: common/zmqclient.cpp [hunk 4]
```diff
@@ -186,4 +203,70 @@ void ZmqClient::sendMsg(
...
 }
 
+bool ZmqClient::wait(std::string& dbName,
+                     std::string& tableName,
+                     std::vector<std::shared_ptr<KeyOpFieldsValuesTuple>>& kcos,
+                     std::vector<char>& buffer)
+{
+    SWSS_LOG_ENTER();
+
+    if (!m_oneToOneSync)
```
Comment (mint570) [inline common/zmqclient.cpp:213]: We can remove this check.
I think the original socket type can do receive. (We change it for performance purposes.)

File: common/zmqclient.cpp (hunk 1 above)
Comment (divyagayathri-hcl) [reply common/zmqclient.cpp:82]: Done.

File: common/zmqclient.h (hunk 3 above)
Comment (divyagayathri-hcl) [reply common/zmqclient.h:52]: Done

File: common/zmqclient.cpp (hunk 4 above)
Comment (divyagayathri-hcl) [reply common/zmqclient.cpp:213]: Done

File: common/zmqclient.h (hunk 2 above)
Comment (divyagayathri-hcl) [reply common/zmqclient.h:15]: Done

**Review by mint570 — DISMISSED** [review]

**Pull request closed without merging**